Documentation/PLUGIN.md

```diff
@@ -182,6 +182,22 @@ exposed via public API, so even if `ImplementationOnlyImports` is set to `true`,
 this will only work if the `Visibility` is set to `internal`.
 
 
+##### Generation Option: `UseAccessLevelOnImports` - imports preceded by a visibility modifier (`public`, `package`, `internal`)
+
+By default, the code generator does not precede any imports with a visibility modifier.
+You can change this with the `UseAccessLevelOnImports` option:
+
+```
+$ protoc --swift_opt=UseAccessLevelOnImports=[value] --swift_out=. foo/bar/*.proto mumble/*.proto
+```
+
+The possible values for `UseAccessLevelOnImports` are:
+
+* `false` (default): Generates plain import directives without a visibility modifier.
+* `true`: Imports of internal dependencies and any modules defined in the module
```

Comment:
I am wondering if we should make the default for Swift 6 be true

Reply:
I would just change the default based on with what version the protoc plugin is compiled with. Since going forward we should have access level on imports by default

Reply:
> We've never had an option option to tell us what version of Swift we're generating for.

That's why I've just set it to `false` for now, it could be changed to `true` once Swift 6 is released.
But we can go with what Franz proposed:

> I would just change the default based on with what version the protoc plugin is compiled with. 

In that case if one still wants to generate without access imports they can explicitly set it through the option.
It's just that the default will automatically change which I think is convenient.

Reply:
I think he's suggesting something like:

```swift
var accessLevelOnImports: Bool? = // .. fetch the option

#if swift(<=5.10)
accessLevelOnImport = accessLevelOnImports ?? false
#else
accessLevelOnImport = accessLevelOnImports ?? true
#endif
```

Note: that was typed directly into github so no claim I'm getting things right, but that should be the general idea.

Reply:
That's also how I understood it

Reply:
I don't have a strong opinion on this. However FWIW, Swift 6 language mode doesn't automatically default imports to internal. As the [SE proposal states](https://github.com/swiftlang/swift-evolution/blob/main/proposals/0409-access-level-on-imports.md#proposed-solution):
>To balance source compatibility and best practices, an import without explicit access level has an implicit access level of public in Swift 5 and Swift 6. It will be internal in a future language mode.